File: February 23/14 Feb ❤️/67. Add Binary LC.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        sum=""
        bitsum=0
        carry=0
        length_a=len(a)-1
        length_b=len(b)-1

        while( length_a >=0 and length_b>=0):
            bitsum = (int(a[length_a]) + int(b[length_b])-int('0')*2 +carry)
            sum+=str(int(bitsum%2))
            carry=bitsum/2
            length_a-=1
            length_b-=1
        print("first loop:"+sum)
        while(length_a>=0):
            bitsum = (int(a[length_a])-int('0') +carry)
            sum+=str(int(bitsum%2))
            carry=bitsum/2
            length_a-=1
        print("Second loop:"+sum)
        while(length_b>=0):
            bitsum = (int(b[length_b])-int('0') +carry)
            sum+=str(int(bitsum%2))
            carry=bitsum/2
            length_b-=1
        print("third loop:"+sum)
        if int(carry)>0:
            sum+=str(int(carry))
        print("forth loop:"+sum)
        return sum[::-1]
        

        

File: February 23/14 Feb ❤️/67. Add Binary LC.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string sum="";
        int bit_a,bit_b,bit_sum,carry=0;
        while(a.size() && b.size())
        {
            bit_a=a.back();
            bit_b=b.back();
            bit_sum=(bit_a-'0'+bit_b-'0'+carry);
            sum.push_back((char)('0'+bit_sum%2));
            carry=bit_sum/2;
            a.pop_back();
            b.pop_back();
        }

        while(a.size())
        {
            bit_a=a.back();
            bit_sum=bit_a-'0'+carry;
            sum.push_back((char)('0'+bit_sum%2));
            carry=bit_sum/2;
            a.pop_back();
        }
        while(b.size())
        {
            bit_b=b.back();
            bit_sum=bit_b-'0'+carry;
            sum.push_back((char)('0'+bit_sum%2));
            carry=bit_sum/2;
            b.pop_back();
        }
        if(carry)
            sum.push_back((char)('0'+carry));

        reverse(sum.begin(), sum.end());
        return sum;

    }
};
